Yosemite/Yosemite/Stores/InboxNotesStore.swift

```diff
@@ -91,6 +93,24 @@ private extension InboxNotesStore {
         }
     }
 
+    /// Dismiss all `InboxNote`s.
+    /// This marks all notifications is_deleted field to true and the inbox notes will be deleted locally.
+    ///
+    func dismissAllInboxNotes(siteID: Int64,
+                              completion: @escaping (Result<Bool, Error>) -> ()) {
+        remote.dismissAllInboxNotes(for: siteID) { [weak self] result in
+            switch result {
+            case .failure(let error):
+                completion(.failure(error))
+
+            case .success(_):
+                self?.deleteStoredInboxNotesInBackground(siteID: siteID, onCompletion: {
```

Comment:
nit: I wonder if it makes sense to only delete notes with IDs returned in the remote response in https://github.com/woocommerce/woocommerce-ios/pull/6257, in case there is ever a change on the server side where some notes can't be deleted 🤔

Reply:
In theory, if some notes can't be deleted, I expect an error from the API, not only the notes deleted. :nod: